src/UserService.Business/Commands/User/CreateUserCommand.cs

```diff
@@ -48,57 +46,66 @@ public class CreateUserCommand : ICreateUserCommand
 
     #region private methods
 
-    private void ChangeUserDepartment(Guid departmentId, Guid userId, List<string> errors)
+    private void EditCompanyEmployee(Guid? departmentId, Guid? positionId, Guid? officeId, Guid userId, List<string> errors)
     {
-      string errorMessage = $"Can't assign user {userId} to the department {departmentId}. Please try again later.";
-      const string logMessage = "Can't assign user {UserId} to the department {DepartmentId}.";
+      string departmentErrorMessage = $"Cannot assign position to user. Please try again later.";
+      string positionErrorMessage = $"Cannot assign department to user. Please try again later.";
+      string officeErrorMessage = $"Cannot assign office to user. Please try again later.";
+      const string departmentLogMessage = "Cannot assign department {departmentId} to user with id {UserId}.";
+      const string positionLogMessage = "Cannot assign position {positionId} to user with id {UserId}.";
+      const string officeLogMessage = "Cannot assign office {officeId} to user with id {UserId}.";
+      const string logMessage = "Cannot edit company employee info for user witd id {UserId}.";
 
       try
       {
-        var response = _rcDepartment.GetResponse<IOperationResult<bool>>(
-          IChangeUserDepartmentRequest.CreateObj(
+        IOperationResult<(bool department, bool position, bool office)> response =
+          _rcEditCompanyEmployee.GetResponse<IOperationResult<(bool department, bool position, bool office)>>(
+          IEditCompanyEmployeeRequest.CreateObj(
             userId,
-            departmentId,
-            _httpContextAccessor.HttpContext.GetUserId())).Result;
-        if (!response.Message.IsSuccess || !response.Message.Body)
+            _httpContextAccessor.HttpContext.GetUserId(),
+            departmentId: departmentId,
+            positionId: positionId,
+            officeId: officeId)).Result.Message;
+
+        if (!response.IsSuccess)
         {
-          _logger.LogWarning(logMessage, userId, departmentId);
+          _logger.LogWarning(logMessage, userId);
 
-          errors.Add(errorMessage);
+          errors.Add(departmentErrorMessage);
+          errors.Add(positionErrorMessage);
+          errors.Add(officeErrorMessage);
```

Comment:
maybe it better if you add the only message of department/position/office if request the entity != null?
if(department != null) {errors.Add(departmentErrorMessage)}.......

Reply:
fixed